pkg/autostart/autostart.go

```diff
@@ -0,0 +1,137 @@
+// Package autostart manage start at login unit files for darwin/linux
+package autostart
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lima-vm/lima/pkg/textutil"
+)
+
+//go:embed lima-vm@INSTANCE.service
+var systemdTemplate string
+
+//go:embed io.lima-vm.autostart.INSTANCE.plist
+var launchdTemplate string
+
+// CreateStartAtLoginEntry respect host OS arch and create unit file
+func CreateStartAtLoginEntry(hostOS, instName, workDir string) error {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	tmpl, err := renderTemplate(hostOS, workDir, instName, os.Executable)
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(unitPath, tmpl, 0o644); err != nil {
+		// Early return if error is unexpected
+		if !errors.Is(err, os.ErrExist) && !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		// Unit file exists, return nil
+		if errors.Is(err, os.ErrExist) {
+			return nil
+		}
+		// .config/systemd folder might be unavailable
+		if errors.Is(err, os.ErrNotExist) && hostOS == "linux" {
+			if err := os.MkdirAll(filepath.Dir(unitPath), os.ModePerm); err != nil {
```

Comment:
Can we do this create all directories on os.stat failure itself. This way a single writeFile should be fine

Reply:
I would unconditionally call `MkdirAll` before `WriteFile`, just to simplify the logic. The call is cheap, and we only do it once. so no need trying to be clever.